types/lodash/fp/index.d.ts

```diff
@@ -0,0 +1,618 @@
+         * two unusual capabilities. First, its arguments needn't be provided one at a time.
+         */
+        curryN(length: number, fn: (...args: any[]) => any): Function;
+
+        /**
+         * Returns a curried equivalent of the provided function. The curried function has two unusual capabilities.
+         * First, its arguments needn't be provided one at a time.
+         */
+        curry<T1, T2, TResult>(fn: (a: T1, b: T2) => TResult): CurriedFunction2<T1, T2, TResult>
+        curry<T1, T2, T3, TResult>(fn: (a: T1, b: T2, c: T3) => TResult): CurriedFunction3<T1, T2, T3, TResult>
+        curry<T1, T2, T3, T4, TResult>(
+            fn: (
+                a: T1, b: T2, c: T3,
+                d: T4
+            ) => TResult
+        ): CurriedFunction4<T1, T2, T3, T4, TResult>
+        curry<T1, T2, T3, T4, T5, TResult>(
+            fn: (
+                a: T1, b: T2, c: T3, d: T4,
+                e: T5
+            ) => TResult
+        ): CurriedFunction5<T1, T2, T3, T4, T5, TResult>
+        curry(fn: Function): Function
+
+        /**
+         * Creates a function that returns value.
+         *
+         * @param value The value to return from the new function.
+         * @return Returns the new function.
+         */
+        constant<T>(value: T): () => T;
+
+        /**
+         * A no-operation function that returns undefined regardless of the arguments it receives.
+         *
+         * @return undefined
+         */
+        noop(...args: any[]): void;
+
+        /**
+         * Creates a function that invokes the method at path on a given object. Any additional arguments are provided
+         * to the invoked method.
+         *
+         * @param path The path of the method to invoke.
+         * @param args The arguments to invoke the method with.
+         * @return Returns the new function.
+         */
+        method<TObject, TResult>(path: string, ...args: any[]): (object: TObject) => TResult;
+
+        /**
+         * Creates an object composed of the picked `object` properties.
+         *
+         * @static
+         * @memberOf _
+         * @category Object
+         * @param {...(string|string[])} [props] The property names to pick, specified
+         *  individually or in arrays.
+         * @param {Object} object The source object.
+         * @returns {Object} Returns the new object.
+         */
+        pick<TResult extends {}, T extends {}>(predicate: string[], object: T): TResult;
+        pick<TResult extends {}, T extends {}>(predicate: string[]): (object: T) => TResult;
+
+        /**
+         * Checks if `value` is `null` or `undefined`.
+         *
+         * @static
+         * @memberOf _
+         * @category Lang
+         * @param {*} value The value to check.
+         * @returns {boolean} Returns `true` if `value` is nullish, else `false`.
+         */
+        isNil(value?: any): boolean;
```

Comment:
Should this be `isNil(value?: any): value is null | undefined` ?

Reply:
It also doesn't make sense for the parameter to be optional here, because `_.isNil()` doesn't perform a useful function (always returns false).